axelor-account/src/main/java/com/axelor/apps/account/db/repo/MoveManagementRepository.java

```diff
@@ -184,7 +184,12 @@ public Move save(Move move) {
               && CollectionUtils.isNotEmpty(moveLine.getInvoiceTermList())) {
             if (moveLine.getInvoiceTermList().stream()
                 .allMatch(invoiceTermService::isNotReadonly)) {
-              moveLine.clearInvoiceTermList();
+              if (moveLine.getInvoiceTermList().stream()
```

Comment:
why do you add a if and not just do:
```
if(moveLine.getInvoiceTermList().stream()
                .allMatch(invoiceTermService::isNotReadonly) 
&& moveLine.getInvoiceTermList().stream()
.filter(invoiceTerm -> invoiceTerm.getIsHoldBack())
                      .count()
                  == 0)
```
Because we should also throw the AxelorException if there is an InvoiceTerm with isHoldBack = true

Reply:
Fixed